src/server/pfs/server/driver.go

```diff
@@ -675,6 +671,66 @@ func (d *driver) deleteProject(ctx context.Context, txnCtx *txncontext.Transacti
 	return nil
 }
 
+// Set child.ParentCommit (if 'parent' has been determined) and write child to parent's ChildCommits
+func (d *driver) linkParent(txnCtx *txncontext.TransactionContext, child *pfs.CommitInfo, parent *pfs.Commit, needsFinishedParent bool) error {
+	if parent == nil {
+		return nil
+	}
+	// Resolve 'parent' if it's a branch that isn't 'branch' (which can
+	// happen if 'branch' is new and diverges from the existing branch in
+	// 'parent').
+	// Clone the parent proto because resolveCommit will modify it.
+	parent = proto.Clone(parent).(*pfs.Commit)
+	parentCommitInfo, err := d.resolveCommit(txnCtx.SqlTx, parent)
+	if err != nil {
+		return errors.Wrapf(err, "parent commit not found")
+	}
+	// fail if the parent commit has not been finished
+	if needsFinishedParent && parentCommitInfo.Finishing == nil {
+		return errors.Errorf("parent commit %s has not been finished", parentCommitInfo.Commit)
+	}
+	child.ParentCommit = parentCommitInfo.Commit
+	parentCommitInfo.ChildCommits = append(parentCommitInfo.ChildCommits, child.Commit)
+	// Note: error is emitted if parent.ID is a missing/invalid branch OR a
```

Comment:
This should be detected by the resolve commit above right?

Reply:
yep! removing comment